### PR TITLE
Setup Autogenerated Client

### PR DESCRIPTION
**Pull request by alexn400**
## Summary of Changes Made
*Brief summary of changes made*
- [x] Setup the auto generated client from the openapi spec

### CODE REVIEW

**Review by kche356 — APPROVED** [review]
aight bet

File: frontend/package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "@emotion/react": "^11.8.1",
     "@emotion/styled": "^11.8.1",
     "@mui/material": "^5.4.4",
+    "@openapitools/openapi-generator-cli": "^2.4.26",
```
Comment (PeterGoedeke) [inline frontend/package.json:9]: Should this be a devDependency?

File: frontend/scripts/generateSwaggerClient.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  fs.mkdirSync(outDir);
+}
+
+const specURL = './swagger.yml';
+
+console.log(`Generating client API from spec at ${specURL}`);
+
+// Generate the client api
+exec(
+  `openapi-generator-cli generate -g typescript-axios --additional-properties=useSingleRequestParameter=true,withoutPrefixEnums=true, -i ${specURL} -o ${tempDir}`,
```
Comment (PeterGoedeke) [inline frontend/scripts/generateSwaggerClient.ts:21]: Could we do this by just putting the shell script directly as a script in package.json?

Comment (alexn400) [reply frontend/scripts/generateSwaggerClient.ts:21]: Could technically do that but itd get kinda ugly with all the other operations like creating and deleting folders and also cant have one solution thatll work on all Operating Systems

**Pull request merged**